gcloud-java-storage/src/test/java/com/google/gcloud/storage/ITStorageTest.java

```diff
@@ -168,13 +166,29 @@ public void testUpdateBlob() {
     assertNotNull(storage.create(blob));
     BlobInfo updatedBlob = storage.update(blob.toBuilder().contentType(CONTENT_TYPE).build());
     assertNotNull(updatedBlob);
-    assertEquals(blob.bucket(), updatedBlob.bucket());
-    assertEquals(blob.name(), updatedBlob.name());
+    assertEquals(blob.blobId(), updatedBlob.blobId());
     assertEquals(CONTENT_TYPE, updatedBlob.contentType());
     assertTrue(storage.delete(bucket, blobName));
   }
 
   @Test
+  public void testUpdateBlobReplaceMetadata() {
+    String blobName = "test-update-blob-replace-metadata";
+    BlobInfo blob = BlobInfo.builder(bucket, blobName)
+        .contentType(CONTENT_TYPE)
+        .metadata(ImmutableMap.of("k1", "a"))
+        .build();
+    assertNotNull(storage.create(blob));
```

Comment:
Can we also add a test for these 2 use-cases (so we can tell if something has changed):
1) merge
2) clear individuals metadata attributes.

Actually (2) let me think that we have a bug.... as it should be possible to clear individual metadata attributes by setting their value to null, however we are not allowing it as we are copying the map into an `ImmutableMap` that disallows nulls...